### PR TITLE
Redeem Badges

### PR DESCRIPTION
**Pull request by hpereira98**
# Description
- Screen added for each badge, allowing the badge to be redeemed by reading a QR Code using the camera or from the gallery. External libs were used to adapt the text size to the screen and for the QR Codes. Also, permissions to use the camera and the local storage were added to the AndroidManifest.xml.

# Features
- Selecting badge from grid
- Redeeming badge with QR Code
- Request status presented in screen

### CODE REVIEW

File: lib/screens/Badge/index.dart [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+      stText = "Invalid link.";
+      stColor = Colors.red;
+    }
+
+    setState(() {
+      isLoading = false;
+      statusText = stText;
+      statusColor = stColor;
+    });
+  }
```
Comment (BrunoMartins11) [inline lib/screens/Badge/index.dart]: This method is too long. Extract some methods to make maintainability easier.

File: lib/components/badgeGrid.dart [hunk 2]
```diff
@@ -2,26 +2,27 @@ import 'package:flutter/cupertino.dart';
...
-
+class BadgeGrid {
   final List<Badge> badges;
-  final List<Image> images;
 
-  BadgeGrid(this.badges, this.images);
+  BadgeGrid(this.badges);
 
-  List<Card> buildGrid() {
+  List<Card> buildGrid(BuildContext context) {
```
Comment (BrunoMartins11) [inline lib/components/badgeGrid.dart:10]: This change is causing an error on file lib/screens/Home/index.dart line 85.

**Review by BrunoMartins11 — CHANGES_REQUESTED** [review]

File: lib/components/badgeGrid.dart (hunk 2 above)
Comment (hpereira98) [reply lib/components/badgeGrid.dart:10]: I believe this is fixed now.

File: lib/screens/Badge/index.dart (hunk 1 above)
Comment (hpereira98) [reply lib/screens/Badge/index.dart]: I have shortened it.

**Review by BrunoMartins11 — APPROVED** [review]

**Pull request merged**